centreon/www/front_src/src/ResourceAccessManagement/AddEditResourceAccessRule/FormInitialValues/initialValues.ts

```diff
@@ -1,25 +1,62 @@
-import { ResourceAccessRule } from '../../models';
+/* eslint-disable no-param-reassign */
+import { equals, prop } from 'ramda';
 
-export const getEmptyInitialValues = (): ResourceAccessRule => ({
+import {
+  Dataset,
+  DatasetFilter,
+  ResourceAccessRule,
+  ResourceTypeEnum
+} from '../../models';
+
+export const getEmptyInitialValues = (): Omit<ResourceAccessRule, 'id'> => ({
   contactGroups: [],
   contacts: [],
-  datasetFilters: [[{ resourceType: undefined, resources: [] }]],
+  datasetFilters: [[{ resourceType: ResourceTypeEnum.Empty, resources: [] }]],
   description: '',
   isActivated: true,
   name: ''
 });
 
+const formatDatasetFilter = (datasetFilter: DatasetFilter): Array<Dataset> => {
+  let datasets: Array<Dataset> = [];
```

Comment:
Use constant instead
```suggestion
  const datasets: Array<Dataset> = [];
```

Reply:
done